### PR TITLE
Update to Alpine 3.20 and kcov v42

### PR DESCRIPTION
**Pull request by williamdes**
Hello @ko1nksm 
I am suggesting this update that I tested, in order to update the image on the hub.
As my image depends on it https://github.com/sudo-bot/action-shunit2

Let me know if you need me to setup a GitHub workflow to publish to Docker directly.
And also do multi arch builds.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,15 @@
-FROM alpine:3.12 as builder
+FROM alpine:3.20 as builder
 RUN apk add --no-cache build-base cmake ninja python3 \
       binutils-dev curl-dev elfutils-dev
 WORKDIR /root
-ENV KCOV=https://github.com/SimonKagstrom/kcov/archive/v38.tar.gz
+ENV KCOV=https://github.com/SimonKagstrom/kcov/archive/refs/tags/v42.tar.gz
 RUN wget -q $KCOV -O - | tar xz -C ./ --strip-components 1
 RUN mkdir build && cd build \
  && CXXFLAGS="-D__ptrace_request=int" cmake -G Ninja .. \
```
Comment (williamdes) [inline Dockerfile:8]:
```suggestion
 && cmake -G Ninja .. \
```
For the next release https://github.com/SimonKagstrom/kcov/pull/443

**Comment by williamdes** [comment]
This Docker image is not needed anymore as upstream has moved to Alpine and is multi arch
https://github.com/SimonKagstrom/kcov/pull/441

**Comment by ko1nksm** [comment]
Sorry for the late reply. I had forgotten what I was using this repository for. This repository is part of the ShellSpec project and perhaps your https://github.com/sudo-bot/action-shunit2 should not depend on this repository.

This Dockerfile uses Docker hub's automated build to build Docker images, and has been broken since Docker stopped providing it for free. I have not decided what to do with this repository at this time, but I will probably discontinue it.

Thanks for the change at upstream. It is helpful for me as well.

**Comment by williamdes** [comment]
> This repository is part of the ShellSpec project and perhaps your https://github.com/sudo-bot/action-shunit2 should not depend on this repository.

Indeed, I moved to kcov/kcov

> Thanks for the change at upstream. It is helpful for me as well.

Maybe you can drop this image when the next release of kcov is out ?
Since the two images will be nearly identical :)

or can you push a build to https://hub.docker.com/r/shellspec/kcov

**Comment by ko1nksm** [comment]
I believe I was using this image for stable ShellSpec CI without depending on upstream changes. I may have wanted to save the time (only 1 min) it takes to build the image. I have looked into it, but it is not used for anything other than CI and there is no documentation. So that means it is an internal use.

At the moment I would like to drop this image in the future, but that will happen at a time convenient for ShellSpec, regardless of the next release of kcov. It is not clear when that will be. In the meantime, I may push a new image. I am interested in a GitHub workflow that does multi arch builds, but  I will do that in the [ShellSpec repository](https://github.com/shellspec/shellspec), not this repository.

**Comment by ko1nksm** [comment]
I have reduced one dependency from ShellSpec to shellspec/kcov docker image.
https://github.com/shellspec/shellspec/commit/bf5c49ff630c2b44acf88bb415abeffe7cedf6b0

There is probably one more left. I will eventually remove that too, but it will take some time.
After that, this repository will no longer be necessary for me.

I'll leave it for a little while in case someone else is using it, but eventually
I will probably remove the entire repository. If there are any problems, please let me know.

P.S. I will leave it as is at least until the next kcov version (v43?) is released.

**Comment by williamdes** [comment]
> P.S. I will leave it as is at least until the next kcov version (v43?) is released.

Yes because the image kcov/kcov:latest has some bugs to fix on bash reporting. Something like that

**Comment by williamdes** [comment]
When this is done, you will be able to remove more lines of your Dockerfile by just copying the built binary into your Docker image directly from kcov/kcov:v43

**Pull request merged**